File: frontend/src/types/invoice.ts
export interface InvoiceDetail {
  id: number
  invoice_id: number
  biz_type?: string | null
  item?: string | null
  qty: number
  amount: number
  tax_rate: number
  tax: number
  total: number
}

export interface Invoice {
  id: number
  invoice_code?: string | null
  invoice_type: string
  code?: string | null
  no: string
  invoice_date?: string | null
  buyer_name?: string | null
  buyer_tax_no?: string | null
  seller_name: string
  seller_tax_no?: string | null
  seller_address_phone?: string | null
  seller_bank_account?: string | null
  account?: string | null
  certify: string
  remark?: string | null
  reimbursement_bill_id?: number | null
  purchase_requisition_item_id?: number | null
  attachment_path?: string | null
  route_info?: string | null
  traveler?: string | null
  created_at: string
  details: InvoiceDetail[]
}

// 发票箱（收口/暂存）记录
export interface InvoiceInbox {
  id: number
  filename: string
  storage_path: string
  source: string // upload | box
  duplicated?: boolean // P1 去重：本次上传与箱中已有同票重复时置 true
  extracted_json?: string | null
  status: string // pending | recognized | linked | error
  linked_doc_type?: string | null // reimburse | purchase
  linked_doc_id?: number | null
  verify_result?: string | null // none | real | fake | abnormal
  verify_note?: string | null
  created_at: string
  recognized_at?: string | null
  linked_at?: string | null
}

export interface InvoiceCreatePayload {
  invoice_type: string
  code?: string | null
  no: string
  invoice_date?: string | null
  buyer_name?: string | null
  buyer_tax_no?: string | null
  seller_name: string
  seller_tax_no?: string | null
  seller_address_phone?: string | null
  seller_bank_account?: string | null
  account?: string | null
  certify: string
  remark?: string | null
  reimbursement_bill_id?: number | null
  purchase_requisition_item_id?: number | null
  attachment_path?: string | null
  route_info?: string | null
  traveler?: string | null
  details: {
    biz_type?: string | null
    item?: string | null
    qty: number
    amount: number
    tax_rate: number
    tax: number
    total: number
  }[]
}

File: backend/app/schemas/invoice.py
"""发票管理 Pydantic 模型：请求/响应结构。"""
from datetime import date, datetime
from decimal import Decimal
from typing import List, Optional

from pydantic import BaseModel, ConfigDict


class InvoiceDetailBase(BaseModel):
    biz_type: Optional[str] = None
    item: Optional[str] = None
    qty: Decimal = Decimal(1)
    amount: Decimal = Decimal(0)
    tax_rate: Decimal = Decimal(0)
    tax: Decimal = Decimal(0)
    total: Decimal = Decimal(0)


class InvoiceDetailCreate(InvoiceDetailBase):
    pass


class InvoiceDetailRead(InvoiceDetailBase):
    id: int
    invoice_id: int
    model_config = ConfigDict(from_attributes=True)


class InvoiceBase(BaseModel):
    invoice_type: str = "增值税专用发票"
    code: Optional[str] = None
    no: str
    invoice_date: Optional[date] = None
    buyer_name: Optional[str] = None
    buyer_tax_no: Optional[str] = None
    seller_name: str
    seller_tax_no: Optional[str] = None
    seller_address_phone: Optional[str] = None
    seller_bank_account: Optional[str] = None
    account: Optional[str] = None
    certify: str = "none"
    remark: Optional[str] = None
    reimbursement_bill_id: Optional[int] = None
    attachment_path: Optional[str] = None
    route_info: Optional[str] = None
    traveler: Optional[str] = None


class InvoiceCreate(InvoiceBase):
    details: List[InvoiceDetailCreate] = []


class InvoiceUpdate(BaseModel):
    invoice_type: Optional[str] = None
    code: Optional[str] = None
    no: Optional[str] = None
    invoice_date: Optional[date] = None
    buyer_name: Optional[str] = None
    buyer_tax_no: Optional[str] = None
    seller_name: Optional[str] = None
    seller_tax_no: Optional[str] = None
    seller_address_phone: Optional[str] = None
    seller_bank_account: Optional[str] = None
    account: Optional[str] = None
    certify: Optional[str] = None
    remark: Optional[str] = None
    reimbursement_bill_id: Optional[int] = None
    attachment_path: Optional[str] = None
    route_info: Optional[str] = None
    traveler: Optional[str] = None
    details: Optional[List[InvoiceDetailCreate]] = None


class InvoiceRead(InvoiceBase):
    id: int
    created_at: datetime
    details: List[InvoiceDetailRead] = []
    model_config = ConfigDict(from_attributes=True)
